File: bni-anchor-checkin-backend/src/test/kotlin/com/example/bnianchorcheckinbackend/GuestLtTermTest.kt
package com.example.bnianchorcheckinbackend

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Assertions.assertNull
import org.junit.jupiter.api.Test

class GuestLtTermTest {
    @Test
    fun `dates before October 2026 are term 2`() {
        assertEquals(2, GuestLtTerm.number("2026-03-12"))
        assertEquals(2, GuestLtTerm.number("2026-09-30"))
    }

    @Test
    fun `October 2026 through March 2027 is term 3`() {
        assertEquals(3, GuestLtTerm.number("2026-10-01"))
        assertEquals(3, GuestLtTerm.number("2027-03-31"))
    }

    @Test
    fun `term advances every six months after March 2027`() {
        assertEquals(4, GuestLtTerm.number("2027-04-01"))
        assertEquals(5, GuestLtTerm.number("2027-10-01"))
    }

    @Test
    fun `missing date has no term`() {
        assertNull(GuestLtTerm.number(null))
        assertNull(GuestLtTerm.number(""))
    }
}

File: bni-anchor-checkin/src/__tests__/guestLtTerm.test.ts
import { describe, expect, it } from "vitest";
import { guestLtTermLabel } from "../lib/guestLtTerm";

describe("guestLtTermLabel", () => {
  it("labels dates before October 2026 as the 2nd LT", () => {
    expect(guestLtTermLabel("2026-03-12")).toBe("第 2 屆 LT");
    expect(guestLtTermLabel("2026-05-21")).toBe("第 2 屆 LT");
    expect(guestLtTermLabel("2026-09-30")).toBe("第 2 屆 LT");
  });

  it("labels October 2026 through the end of March as the 3rd LT", () => {
    expect(guestLtTermLabel("2026-10-01")).toBe("第 3 屆 LT");
    expect(guestLtTermLabel("2026-12-17")).toBe("第 3 屆 LT");
    expect(guestLtTermLabel("2027-03-31")).toBe("第 3 屆 LT");
  });

  it("advances one term every six months after March 2027", () => {
    expect(guestLtTermLabel("2027-04-01")).toBe("第 4 屆 LT");
    expect(guestLtTermLabel("2027-09-30")).toBe("第 4 屆 LT");
    expect(guestLtTermLabel("2027-10-01")).toBe("第 5 屆 LT");
    expect(guestLtTermLabel("2028-03-31")).toBe("第 5 屆 LT");
  });

  it("returns null when the date is missing", () => {
    expect(guestLtTermLabel(undefined)).toBeNull();
    expect(guestLtTermLabel("")).toBeNull();
  });
});
